### PR TITLE
(DOCSP-27962) Document React Native Polyfills for RN .watch() method

### PR DESCRIPTION
**Pull request by krollins-mdb**
### Jira

- https://jira.mongodb.org/browse/DOCSP-27962

### Staged Changes

- [Query MongoDB](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27962/sdk/react-native/app-services/query-mongodb/#real-time-change-notifications)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+let higherScopedVenusFlyTrap;
+const higherScopedPlantId = new Realm.BSON.ObjectID('5f87976b7b800b285345a8b4');
+
+// Note: have to create and wait for this promise because
+// otherwise the test fails and exits before all the network connection notifications
+// have resolved.
+let promiseResolve;
+const promise = new Promise(function (resolve) {
+  promiseResolve = resolve;
+});
```
Comment (mongodben) [inline examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:18]: yay glad this worked!

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  }, [user, watchForAllChanges]);
+
+  const watchForAllChanges = async plants => {
+    // Watch for changes to the plants collection
+    for await (const change of plants.watch()) {
+      switch (change.operationType) {
+        case 'insert': {
+          const {documentKey, fullDocument} = change;
+          // ... do something with the change information.
+          console.log(`new document: ${documentKey}`, fullDocument); // :remove:
```
Comment (mongodben) [inline examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:62]: why are you logging this and removing the logs?

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  beforeAll(async () => {
+    const app = Realm.App.getApp(APP_ID);
+
+    await app.logIn(Realm.Credentials.anonymous());
+
+    plants =
+      app.currentUser.mongoClient('mongodb-atlas').db('example').collection <
+      Plant >
+      'plants';
```
Comment (mongodben) [inline examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:105]: this doesn't look correct

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    promiseResolve(true); // :remove:
+  };
+
+  // ... rest of component
+
+  return <></>; // :remove:
+}
+// :snippet-end:
+
+describe('MongoDB Watch', () => {
```
Comment (mongodben) [inline examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:94]: how do these tests validate that the data was watched? it looks to me like they're just testing the operations within the tests.

File: source/sdk/react-native/app-services/query-mongodb.txt [hunk 5]
```diff
@@ -357,12 +357,29 @@ document in the collection is added, modified, or deleted. Each notification
...
+</reference/change-events/>` for operations as they occur.
 
-   ``collection.watch()`` returns an :mdn:`async generator
-   <Web/JavaScript/Reference/Statements/for-await...of#Iterating_over_async_generators>`
-   that allows you to asynchronously pull :manual:`change events
-   </reference/change-events/>` for operations as they occur.
+``collection.watch()`` requires some set up to work with a React Native client
+app. You need the :npm:`react-native-polyfill-globals </packagereact-native-polyfill-globals/>`
+and :npm:`@babel/plugin-proposal-async-generator-functions 
+</package/@babel/plugin-proposal-async-generator-functions>` packages.
```
Comment (mongodben) [inline source/sdk/react-native/app-services/query-mongodb.txt:368]:
```suggestion
app. To watch a collection for changes, you must first install the :npm:`react-native-polyfill-globals </packagereact-native-polyfill-globals/>`
and :npm:`@babel/plugin-proposal-async-generator-functions 
</package/@babel/plugin-proposal-async-generator-functions>` packages.
```

File: source/sdk/react-native/app-services/query-mongodb.txt [hunk 6]
```diff
@@ -357,12 +357,29 @@ document in the collection is added, modified, or deleted. Each notification
...
+**To use collection.watch():** 
+
+#. Install dependencies.
+
+   .. code-block:: bash
+   
+      npm install react-native-polyfill-globals text-encoding
+      npm install --save-dev @babel/plugin-proposal-async-generator-functions
+
+#. Import polyfills in ``index.js``.
```
Comment (mongodben) [inline source/sdk/react-native/app-services/query-mongodb.txt:379]: must it be in index.js or can it just be on a higher scope than where you use the polyfills?

File: source/sdk/react-native/app-services/query-mongodb.txt [hunk 7]
```diff
@@ -372,11 +389,27 @@ the operation that caused the event.
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+With the ``@realm/react`` package, make sure you have 
+:ref:`configured user authentication <react-native-configure-authentication>`
+for your app.
+
 To watch for all changes in a collection, call :js-sdk:`collection.watch()
-<Realm.MongoDBCollection.html#watch>` with no arguments:
+<Realm.MongoDBCollection.html#watch>` with no arguments. This call must be inside
+of ``UserProvider`` with an authenticated user.
```
Comment (mongodben) [inline source/sdk/react-native/app-services/query-mongodb.txt:398]:
```suggestion
<Realm.MongoDBCollection.html#watch>` with no arguments. This call must be
wrapped by a ``UserProvider`` with an authenticated user.
```

**Review by mongodben — COMMENTED** [review]
nice start. couple small copy edits and some structural questions for the example

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx (hunk 2 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:62]: I could leave it in, I suppose. The previous commented line was intended to server as a generic catch all.

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx (hunk 3 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:105]: Weird that this ran at all though. The test passed. Gonna have to look into that.

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx (hunk 4 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:94]: Excellent point. I'm refactoring to look at that.

File: examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx (hunk 2 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/app-services/mongodb-watch.test.jsx:62]: I've removed the console logs in my next commit.

File: source/sdk/react-native/app-services/query-mongodb.txt (hunk 6 above)
Comment (krollins-mdb) [reply source/sdk/react-native/app-services/query-mongodb.txt:379]: Ah, good point. Should be able to import anywhere that's a higher scope than the polyfills themselves.

**Pull request merged**